iothub_client/tests/iothubclient_fuzz_amqp/iothubclient_fuzz.c

```diff
@@ -0,0 +1,685 @@
+        size_read = fread(&count_xor, 1, 1, fp);
+        size_read = fread(file_buffer, 1, size, fp);
+        fclose(fp);
+    }
+    else
+    {
+        exit(-1);
+    }
+    return size_read;
+}
+
+int tlsio_fuzz_send(CONCRETE_IO_HANDLE tls_io, const void* buffer, size_t size, ON_SEND_COMPLETE on_send_complete, void* callback_context)
+{
+    (void)tls_io;
+    static int iteration = 0;
+
+    size_t size_read;
+    unsigned char file_buffer[4096];
+
+    switch (iteration)
+    {
+        case 0: // client SASL tunnel request
+            // server SASL tunnel response
+            size_read = load_from_file(0, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            // server SASL Mechanisms
+            size_read = load_from_file(1, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            break;
+
+        case 1: // client SASL Init
+            // server SASL Challenge
+            size_read = load_from_file(2, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            break;
+
+        case 2: // client connection header
+            // server connection header
+            //<- Header (AMQP 0.1.0.0)
+            size_read = load_from_file(3, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            break;
+
+        case 3: // client performative Open
+            // server begin
+            //<- [OPEN]* {DeviceGateway_5ad507ef78724b7386c9c3309fb31014,localhost,65536,8191,240000,NULL,NULL,NULL,NULL,NULL}
+            size_read = load_from_file(4, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            break;
+
+        case 4: // send begin
+            //<- [BEGIN]* {0,1,5000,4294967295,262143,NULL,NULL,NULL}
+            size_read = load_from_file(5, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            break;
+
+        case 5: // send attach $cbs-sender
+            //<- [ATTACH]* {$cbs-sender,0,true,0,0,* {$cbs,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL},* {$cbs,NULL,NULL,NULL,NULL,NULL,NULL},NULL,NULL,NULL,1048576,NULL,NULL,NULL}
+            size_read = load_from_file(6, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            break;
+
+        case 6: // send attach $cbs-receiver
+            //<- [ATTACH]* {$cbs-receiver,1,false,0,0,* {$cbs,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL},* 
+            size_read = load_from_file(8, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            //< -[FLOW] * {0, 5000, 1, 4294967295, 0, 0, 100, 0, NULL, false, NULL}
+            size_read = load_from_file(7, file_buffer, sizeof(file_buffer));
+            received_queue_add(file_buffer, size_read);
+            break;
```

Comment:
just double checking that these are supposed to be out of order

Reply:
yes, solves a timing issue on the packet capture part